src/docs/services/queue.mdx

```diff
@@ -23,12 +23,53 @@ def do_work(kind_of_work, **kwargs):
     # ...
 ```
 
-There's a few key differences we practice:
+There are a few important points:
 
 - The task name _must_ be declared.
+
+  The task name is how celery identifies messages (requests) and which
+  function and worker needs to handle those messages.  If the tasks
+  are not named celery will derive a name from the module and function
+  names which makes the name tied to the location of the code and more
+  brittle for future code maintenance.
+
 - Tasks _must_ accept `\*\*kwargs`` to handle rolling compatibility.
+
+  This ensures tasks will accept any message which happens to be in
+  the queue rather than fail for unknown arguments.  It helps with
+  rolling changes back and forwards, deploys are not instant and
+  messages may be produced with multiple versions of arguments.
+
+  While this allows rolling forwards and backwards without complete
+  task failures, care must still be taken for workers to handle
+  messages with both old and new arguments when changing the
+  arguments.  This does reduce the number of required changes in such
+  a migration a little and gives some more operator flexibility, but
+  message loss because of unknown arguments is still not acceptable.
+
 - Tasks _should_ automatically retry on failure.
 
+- Tasks arguments _should_ be primitive types and small.
+
+  Task arguments are serialised into the message sent across the
+  brokers and the workers need to deserialise them again.  Doing this
+  with complex types is brittle and should be avoided.  E.g. prefer to
+  pass an ID to the task which can be used to load the data from the
+  database rather than the data itself.
+
+  Similarly to keep the message borkers and workers running
+  efficiently serlialising large values into the message results in
+  large messages, large queues and more (de)serialising overheads so
```

Comment:
```suggestion
  efficiently serialising large values into the message results in
  large messages, large queues and more (de)serialising overheads so
```
Had some difficulties with parsing this sentence, perhaps it should be rephrased a bit.